[WIP] Add phonemizer for Belarusian language

alex73 opened the pull request:
@erogol I'm trying to implement phonemizer for Belarusian language. Could you please to take a look ?

Idea is to call java class from python.
3 things below are required only when Belarusian phonemizer really executed:
- jpype1 module
- JAVA_HOME env variable - points to java installation directory
- BEL_FANETYKA_JAR env variable - points to fanetyka.jar from the https://github.com/alex73/Software-Korpus/releases

I'm not sure about test. I wrote it, but it requires installed java and exists fanetyka.jar. I'm not sure that CoquiTTS test environment will have they.

erogol commented on requirements.txt:
```diff
@@ -48,3 +48,5 @@ einops
 #deps for bark
 encodec
+#deps for Belarusian phonemizer
+jpype1
```
This is a big dependency. It is better to keep out of the requirements but warn the user when the phonemizer is called without it being installed.

alex73 replied:
Right. Fixed by change TTS/tts/utils/text/belarusian/phonemizer.py

erogol commented on TTS/tts/utils/text/phonemizers/__init__.py:
```diff
@@ -30,6 +31,7 @@
 DEF_LANG_TO_PHONEMIZER["zh-cn"] = ZH_CN_Phonemizer.name()
 DEF_LANG_TO_PHONEMIZER["ko-kr"] = KO_KR_Phonemizer.name()
 DEF_LANG_TO_PHONEMIZER["bn"] = BN_Phonemizer.name()
+DEF_LANG_TO_PHONEMIZER["be"] = BEL_Phonemizer.name()
```
it is better not to default it to your implementation. Espeak also supports BE. Since it doesn't require Java, I think it is better to default to that.

Maybe you can key your implementation with a different keyword like `be_fanetyka`

erogol replied:
We should also handle the case when java backend is not installed like here -> https://github.com/coqui-ai/TTS/pull/2843

alex73 replied:
Espeak has very bad phonemizer for Belarusian language - no many rules, no stresses, no changes on the border of words. It's not really useful. Since CoquiTTS doesn't have espeak phonemizer for Belarusian language, I don't think it will be good to add it.
I changed TTS/tts/utils/text/belarusian/phonemizer.py for support case when no java installed. Belarusian phonemizer initialized only on the first real call, so, it will not affect on other phonemizers and will not load any java code before real usage.

alex73 commented:
@erogol I updated TTS/tts/utils/text/belarusian/phonemizer.py implementation. I hope it looks better now.

erogol commented:
@alex73 this looks good. Thanks for the changes. I'll test locally and merge.

erogol commented:
@alex73 looks like the tests are failing since the steps are missing. You should probably skip the test if the dips are not installed. Otherwise, looks ready to merge

lexkoro commented:
The latest espeak-ng version does support belarusian

alex73 commented:
> @alex73 looks like the tests are failing since the steps are missing. You should probably skip the test if the dips are not installed. Otherwise, looks ready to merge

Fixed. I added warning to unit test in case of jar is not defined

alex73 commented:
> The latest espeak-ng version does support belarusian

Yes, but it supports only very simple transformation, not any real complex things.

erogol commented:
Thanks for the fix. I'll check again on Monday and merge most probably.

erogol commented:
@alex73 thanks for the PR and patience!!
